### PR TITLE
WIP Delay Regex evaluation for ApiClassExtractor#isLocalClass

### PR DESCRIPTION
**Pull request by paplorinc**
Ad hoc perf test:[ `clean assemble on largeMonolithicJavaProject with rich console `](https://builds.gradle.org/viewType.html?buildTypeId=Gradle_Util_Performance_AdHocPerformanceScenarioLinux&tab=buildTypeStatusDiv&branch_Gradle_Util_Performance=lorinc%2Fclass-extractor-pattern-optimization)

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/api/internal/tasks/compile/ApiClassExtractor.java [hunk 1]
```diff
@@ -93,6 +93,8 @@ private static String packageNameOf(String internalClassName) {
 
     // See JLS3 "Binary Compatibility" (13.1)
     private static boolean isLocalClass(String className) {
```
Comment (paplorinc) [inline subprojects/core/src/main/java/org/gradle/api/internal/tasks/compile/ApiClassExtractor.java]: we could rename this to `isInnerClass`

**Comment by paplorinc** [comment]
Yes, bumped into this multiple times during profiling of different flows.
It's not a relevant bottleneck, but the solution optimizes the general flow, i.e. non-inner classes to avoid heavy regexp use.
Would you like me to create a perf test for it?

**Comment by oehme** [comment]
I'd just like to see at least one flame graph where this is visible. It would be great if you could always attach one for before and after when opening performance related PRs. Makes reviewing so much easier.

**Comment by oehme** [comment]
The flame graph linked above is for a native project, which doesn't use this code path at all. The results are inconclusive as expected.

**Comment by paplorinc** [comment]
Fixed the link. Don't see any relevant difference. I will get back to this one if I see this method in a stack again.

**Pull request closed without merging**